test/client/fixtures/sandbox/node/document-test.js

```diff
@@ -494,23 +495,66 @@ if (!browserUtils.isIE) {
 }
 
 test('an iframe should not contain self-removing scripts after document.close (GH-871)', function () {
-    var iframe = document.createElement('iframe');
+    return createTestIframe()
+        .then(function (iframe) {
+            var iframeDocument = iframe.contentDocument;
 
-    iframe.id = 'test-gh-871';
+            iframeDocument.designMode = 'On';
+            iframeDocument.open();
+            iframeDocument.write('<body style=\"padding: 0; margin: 0; overflow: hidden;\"></body>');
+            iframeDocument.close();
 
-    document.body.appendChild(iframe);
+            var selfRemovingScripts = nativeMethods.querySelectorAll.call(iframeDocument,
+                '.' + SHADOW_UI_CLASSNAME.selfRemovingScript);
 
-    var iframeDocument = iframe.contentDocument;
+            strictEqual(selfRemovingScripts.length, 0);
+        });
+});
 
-    iframeDocument.designMode = 'on';
-    iframeDocument.open();
-    iframeDocument.write('<body style=\"padding: 0; margin: 0; overflow: hidden;\"></body>');
-    iframeDocument.close();
+if (browserUtils.isWebKit) {
```

Comment:
I guess we can drop this instruction.
Test should pass in all supported browsers.